src/coroutine.rs

```diff
@@ -250,8 +299,10 @@ impl<'a, Input, Yield, Return, Stack: stack::Stack>
         self.stack_ptr = None;
 
         let mut input = ManuallyDrop::new(input);
+        Self::init_current(self);
```

Comment:
I don't like this approach because it means that every coroutine resume will need to pay the cost of thread-local-storage access on every switch. This can be very expensive, in particular when built as a shared library since all TLS accesses need to call `__tls_get_address` in the runtime library.

Would it be possible to design this differently so that this cost is opt-in instead?

Reply:
Can you accept adding a `maybe-grow` feature? When the feature is enabled, `Self::init_current(self);` will be executed.

Or, how about adding `maybe_grow:bool` field in `ScopedCoroutine`? When the field value is `true`, `Self::init_current(self);` will be executed.

Which one do you prefer? Or do you have a better design?